File: frontend/src/api/history.ts
import { apiClient, unwrap } from './client'

export interface HistoryMessage {
  id: number
  role: string
  content: string
  latency_ms: number
  created_at: string
}

export interface ConversationSummary {
  id: number
  title: string | null
  channel: string
  total_latency_ms: number
  started_at: string
  message_count: number
}

export interface ConversationDetail extends Omit<ConversationSummary, 'message_count'> {
  messages: HistoryMessage[]
}

export function listHistory() {
  return unwrap<ConversationSummary[]>(apiClient.get('/api/history'))
}

export function getHistoryDetail(id: number) {
  return unwrap<ConversationDetail>(apiClient.get(`/api/history/${id}`))
}

File: backend/app/schemas/history.py
from datetime import datetime

from pydantic import BaseModel


class MessageOut(BaseModel):
    id: int
    role: str
    content: str
    latency_ms: int
    created_at: datetime

    model_config = {"from_attributes": True}


class ConversationOut(BaseModel):
    id: int
    title: str | None
    channel: str
    total_latency_ms: int
    started_at: datetime
    message_count: int


class ConversationDetailOut(BaseModel):
    id: int
    title: str | None
    channel: str
    total_latency_ms: int
    started_at: datetime
    messages: list[MessageOut]
